### ISSUE TITLE
Installation failing on M1 MacBook Air and Python 3.8 via Homebrew 

### ISSUE THREAD

**Issue by casey-robertson** [rcaloras/bashhub-client#97]
Failing install on Mac OS Big Sur, MacBook Air (M1 - Apple Silicon).

Sorry in advance, not a Python expert

```
curl -OL https://bashhub.com/setup && zsh setup
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8746  100  8746    0     0  15157      0 --:--:-- --:--:-- --:--:-- 15131
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   122  100   122    0     0    663      0 --:--:-- --:--:-- --:--:--   663
100   282  100   282    0     0    734      0 --:--:-- --:--:-- --:--:--   734
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5001k  100 5001k    0     0  4614k      0  0:00:01  0:00:01 --:--:-- 10.6M
New python executable in /Users/casey/.bashhub/env/bin/python
ERROR: The executable /Users/casey/.bashhub/env/bin/python is not functioning
ERROR: It thinks sys.prefix is '/Users/casey/.bashhub' (should be u'/Users/casey/.bashhub/env')
ERROR: virtualenv is not compatible with this system or executable
```

**Comment by rcaloras** [comment]
Thanks for sharing @casey-robertson! Doing a quick search and seeing it might be related to the specific Python installation being conda. 
Not sure which version of Python it's finding. If you still have the install directory, can you share this:

```
 /Users/casey/.bashhub/env/bin/python --version
```

Alternatively, If you rerun your install command, I just added some additional logging to highlight the path and you can check the version of the Python executable there. https://github.com/rcaloras/bashhub-client/commit/5c488509a49fb03df8d8e66a79ae7f02fa3cde85

**Comment by manthosh** [comment]
I'm also facing the exact same issue. The output of the additional logging in [`5c48850`](https://github.com/rcaloras/bashhub-client/commit/5c488509a49fb03df8d8e66a79ae7f02fa3cde85) is
```
Using Python path /usr/bin/python3
```

Also when I run 
```
 /Users/manthosh/.bashhub/env/bin/python --version
 ```
 the command gets linked. Somehow the symlink is not working
 
<img width="535" alt="Screenshot 2021-06-23 at 1 13 55 PM" src="https://user-images.githubusercontent.com/6534637/123056818-e91f8400-d424-11eb-9caa-d471cd40a9f1.png">

**Comment by jakoberpf** [comment]
I have/had the same issue and fixed it by removing the virtualenv dependency and using the python module "venv". I am not sure of this is a solution, as I don't know if the "venv" module is present in python by default. #105

**Comment by rcaloras** [comment]
Looking more into this now as I have a bit more free time and also experienced this on a new macbook. @jakoberpf's thanks for sharing the PR! This work around is sufficient if venv is present and already installed. Feel free to patch your setup script with #105. However, experimenting with some linux distributions and venv is not always initially available and needs to be installed with `apt-get`.

I'd like to create a solution in setup that should take into account detecting if venv is present (and use it), and otherwise if it's an older python version to manually install virtualenv (current implementation). Think this should also help address https://github.com/rcaloras/bashhub-client/issues/104.

**Comment by rcaloras** [comment]
Alternatively, considering if it makes sense to just migrate to using PEX for distributing an executable https://github.com/pantsbuild/pex

**Comment by rcaloras** [comment]
I put together a set of changes captured on `2.2.0-dev` and https://github.com/rcaloras/bashhub-client/pull/106. This should update the version of virtualenv that's used during install which I believe should be much more compatible with the latest versions of Python and Macs. I only tested this on a few boxes running versions of Python3 and Python2.

@jakoberpf, @proinsias: Any chance you could give installing this version a try on any systems you were experiencing some issues on? You should be able to test simply by running:

```
curl "https://bashhub.com/setup?version=2.2.0-dev" -o setup && bash setup 2.2.0-dev
```

**Comment by jakoberpf** [comment]
@rcaloras Just ran your install/update command successfully. :-)

**Comment by rcaloras** [comment]
@jakoberpf great and thanks! I'm going to merge this into master and release 2.2.0. Any other fixes or tweaks will address as follow ups.

**Comment by rcaloras** [comment]
Closing should be fixed with the release of 2.2.0.
